packages/frontend-app-api/src/wiring/createApp.test.tsx

```diff
@@ -134,4 +135,92 @@ describe('createApp', () => {
 
     await expect(screen.findByText('Derp')).resolves.toBeInTheDocument();
   });
+
+  it('should log an app', () => {
+    const { rootInstances } = createInstances({
+      config: new MockConfigApi({}),
+      plugins: [],
+    });
+    const root = createExtensionInstance({
+      extension: createExtension({
+        id: 'root',
+        at: '',
+        output: {},
+        factory() {},
+      }),
+      config: undefined,
+      attachments: new Map([['children', rootInstances]]),
+    });
+
+    expect(String(root)).toBe(
```

Comment:
totally a candidate for `toMatchInlineSnapshot` for simplicity, but i presume you prefer the trim to get that first newline? :) ``` expect(`\n${root}`).toMatchInlineSnapshot(... ``` 🤓